File: src/main/java/player/ki/ConstantEvaluation.java
package player.ki;

import board.ChessGameInterface;

/**
 * usefull for Profile purposes (only DynamicEvaluation remains)
 *
 * @author Stephan Schröder
 */

public class ConstantEvaluation implements StaticEvaluationInterface {
    @Override
    public Evaluaded evaluate(ChessGameInterface game, boolean forWhite) {
        return Evaluaded.DRAW;
    }
}

File: src/main/kotlin/voidchess/player/ki/evaluation/ConstantEvaluation.kt
package voidchess.player.ki.evaluation

import voidchess.board.ChessGameInterface

/**
 * useful for Profile purposes (only DynamicEvaluation remains)
 */

class ConstantEvaluation : StaticEvaluationInterface {

    override fun getPrimaryEvaluation(game: ChessGameInterface, forWhite: Boolean) = nullValue

    override fun addSecondaryEvaluation(game: ChessGameInterface, forWhite: Boolean, evaluated: Evaluated) {
        //do nothing (keep the secondary value = 0)
    }

    companion object {
        private val nullValue = Ongoing(0.0)
    }
}
